fcsparser/api.py

```diff
@@ -453,7 +468,7 @@ def dataframe(self):
 
 
 def parse(path, meta_data_only=False, output_format='DataFrame', compensate=False,
-          channel_naming='$PnS', reformat_meta=False):
+          channel_naming='$PnS', reformat_meta=False, data_set=0):
```

Comment:
Would it be worthwhile to modify the response to allow a user to more easily fetch all the data:

Now (user has to invoke `parse` multiple times per file)

```
some_info = parse(file)
num_data_sets = extract_num_data_sets(some_info)

for data_set in range(num_data_sets):
  data_set_data = parse(file, data_set=data_set)
```

Alternatively (user only invokes `parse` once):

```
some_info = parse(file) # <-- gets modified into a dictionary response with one of the keys being a list of data.
some_info['data_sets'] # <-- Holds a list of datasets
```

Reply:
I was not quite sure in which way this could be best implemented to ensure consistent behavior.
In effect returning all data sets would return multiple fcs Objects in a list, as there are no required mutual dependencies in the header and the text segments of the FCS specification.
One solution would be adding an toggle returning a list of parsed data and metadata, although this could be quite confusion in conjunction with the data_set option.

Alternatively I could remove data_set entirely and leave it to the user to make sense of the returned list object, although this could lead to quite a lot of unexpected behavior.
What would you prefer?

Reply:
I say let's go with the current version -- it'll work for most users without API changes and for folks that need to fetch multiple data sets they'll be able to do it.